transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/SourceSetUtils.java

```diff
@@ -75,17 +81,43 @@ private static String getConfigurationNameForSourceSet(SourceSet sourceSet, Conf
    * Adds the provided dependency to the given {@link Configuration}
    */
   static void addDependencyToConfiguration(Project project, Configuration configuration, Object dependency) {
-    configuration.withDependencies(dependencySet -> dependencySet.add(project.getDependencies().create(dependency)));
+    addDependencyToConfiguration(configuration, createDependency(project, dependency), null);
+  }
+
+  /**
+   * Adds the provided dependency {@link Dependency} to the given {@link Configuration}
+   *
+   * @param configuration gradle {@link Configuration}
+   * @param dependency gradle {@link Dependency}
+   * @param excludedPackages excluded packages for external module dependency
+   */
+  static void addDependencyToConfiguration(final Configuration configuration, final Dependency dependency,
+      final @Nullable Set<Map<String, String>> excludedPackages) {
+    configuration.withDependencies(dependencySet -> {
+      if (dependency instanceof ModuleDependency && excludedPackages != null) {
```

Comment:
Should we have an `else if (!(dependency instanceof ModuleDependency) && excludedPackages != null)`?

Maybe main `if (excludedPackages != null)`, and inside it we can do an `if/else` for `(dependency instanceof ModuleDependency) `.

Reply:
will sperate. But, there is no action for non ModuleDependency dependency right now.